src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp

```diff
@@ -0,0 +1,53 @@
+#include "Economy/Expenditures.h"
+#include "Entities/EntityManager.h"
+#include "GameCommands.h"
+#include "Types.hpp"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004B0B50
+    static uint32_t vehiclePassSignal(EntityId headId, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        try
+        {
+            Vehicles::Vehicle train(headId);
+            setPosition(train.veh2->position);
+
+            auto& head = train.head;
+            if (!head->canBeModified())
```

Comment:
there is a check missing for the owner to make sure the company is allowed to modify.

Reply:
Oops, I don't know where that got lost. Will re-instate.